### PR TITLE
Use `yazl` instead of `JSZip`

### PR DESCRIPTION
**Pull request by kevva**
Also adds a test to check whether the zip has been created successfully or not by checking the contents.

### CODE REVIEW

**Comment by kevva** [comment]
Here are some benchmarks using this file https://gist.github.com/kevva/54d8c3d62a2bba256df8:

```
                      Extract all files from a ZIP archive
             192 op/s » jszip
             900 op/s » yazl
```

**Comment by sindresorhus** [comment]
@kevva You told me to wait on merging this, but I can't remember the reason?

Merge conflict also needs to be fixed.

**Comment by kevva** [comment]
It seems to work fine by looking at the tests. I remember trying to extract the file using the archive manager on Ubuntu and it failed for some reason though.

**Comment by sindresorhus** [comment]
@kevva Any open issue on `yazl`?

I don't want to use it if it has bugs as I will have to deal with them and JSZip seems to be working ok.

**Comment by kevva** [comment]
Nope, but I keep getting random `write after end` when testing now so I'll have to look into this further.

**Comment by RickyChien** [comment]
Vote for yazl! 

@kevva I ran into the same issue that my zip cannot be unzip properly ```Error 1 - operation not permitted```by Archive Utility (default on Mac) but worked with The unarchiver or ```unzip```.

**Comment by kevva** [comment]
@sindresorhus, I think this should be fine now.

**Comment by RickyChien** [comment]
https://github.com/thejoshwolfe/yazl/issues/14 

It seems that OS X's issue has been fixed hours ago. It's time to merge this PR!

**Comment by kevva** [comment]
:dancer: More than 4x speed improvement!

File: index.js [hunk 1]
```diff
@@ -42,17 +43,17 @@ module.exports = function (filename, opts) {
...
-			contents: zip.generate({
-				type: 'nodebuffer',
-				compression: opts.compress ? 'DEFLATE' : 'STORE',
-				comment: opts.comment
-			})
-		}));
-
-		cb();
+		zip.end(function () {
+			zip.outputStream.pipe(concatStream(function (data) {
```
Comment (andrewrk) [inline index.js]: you don't have to wait until the end callback to pipe the output stream. in fact, you can skip the zip.end function entirely - it's just to notify you when it knows the final size of the zip file.

File: test.js [hunk 2]
```diff
@@ -1,32 +1,43 @@
 'use strict';
+var fs = require('fs');
 var path = require('path');
 var assert = require('assert');
 var gutil = require('gulp-util');
+var unzip = require('decompress-unzip');
```
Comment (andrewrk) [inline test.js]: thejoshwolfe has https://github.com/thejoshwolfe/yauzl as well, just sayin' :-)

File: index.js (hunk 1 above)
Comment (kevva) [inline index.js]: I do think it's needed. I get timeout issues otherwise when running tests.

Comment (andrewrk) [inline index.js]: oh sorry, you do need to call it. but you don't need to wait for the callback.

File: test.js (hunk 2 above)
Comment (kevva) [inline test.js]: @andrewrk, look what library I'm using in decompress-unzip :).

File: index.js (hunk 1 above)
Comment (andrewrk) [inline index.js]: >Data becomes available in this stream soon after calling one of addFile(), addReadStream(), or 
>addBuffer(). Clients can call pipe() on this stream at any time, such as immediately after getting 
>a new ZipFile instance, or long after calling end().

from https://github.com/thejoshwolfe/yazl#outputstream

Comment (kevva) [inline index.js]: Yeah, makes no difference though since we're only handling one buffer at a time.

File: test.js (hunk 2 above)
Comment (andrewrk) [inline test.js]: hahahaha

**Comment by RickyChien** [comment]
Could we update yazl to version 2.2.1 since it fixed on OS X
See also https://github.com/thejoshwolfe/yazl/issues/14

**Comment by kevva** [comment]
@RickyChien, `^2.1.0` will install `2.2.1`.

**Pull request merged**